src/ToCode/CodeDomArgs.cs

```diff
@@ -69,6 +69,7 @@ public CodeDomArgs(CodeTypeDeclaration rootClass, CodeMemberMethod initMethod)
     /// so should have a capital letter at start. <see langword="false"/> for private members
     /// which should start with a lower case letter.</param>
     /// <returns><paramref name="name"/>.</returns>
+    /// TODO: Consider prefixing @ for reserved keywords, so casing rules can be preserved
```

Comment:
I considered this when I wrote this method but decided against it.

`ICodeGenerator` doesn't distinguish between 'reserved keywords' and simply invalid e.g. starting with a number.

Rather than try to list all the reserved type names myself I went with the simplest solution which was to treat all invalid names the same way which is to prepend with `_`.  

For example:

```
// Valid
public int @new {get;set;}

// Not Valid
public int @32 {get;set;}
```

Reply:
As it turns out, I got a list of all illegal keywords and the associated unit test now has them all. 🙂

But yes, there are some interesting limitations with the roslyn stuff, and also some potentially useful parts we might benefit from, as well.

Reply:
So, looking at this on a larger screen...

I actually prefer prepending with an underscore, anyway, since that follows the .net design guide on private member naming and is what I'm used to in other projects. I actually don't remember why I had that thought when I wrote the TODO haha.

Reply:
Oh yeah! I had another thought about this.

For the same reason I suggested making various things in TGD internal (testing), we should probably have it emit the members as internal, as well, instead of private. I've had to turn a lot of them into internal, manually, when using TGD, so that I can test, but they of course get turned back on next edit.

Maybe we could expose accessibility of members as a choice for the user?